lib/apiservers/engine/backends/archive.go

```diff
@@ -636,12 +643,34 @@ func (rm *ArchiveStreamReaderMap) Close() {
 	rm.prefixTrie.Visit(closeStream)
 }
 
+// try to fake the statpath for path that targets the mountpoint or along the mountpoint
+func tryFakeStatPath(mounts []types.MountPoint, target string) (*types.ContainerPathStat, bool) {
+	isMountPathTarget := false
+
+	for _, mount := range mounts {
+		if strings.HasPrefix(mount.Destination, target) {
```

Comment:
We want to fake the statpath for `/mnt` because any directory along the mount path will be created. The problem is that before the container is started, directory in the mount path are not created yet, so stating it on the container filesystem will fail, even though it's a valid target. It's hard to see with `/mnt` since that already exists. But if the mount path is `/vol/A` then 'docker cp test.txt cid:/vol` is a valid cp command, but stating it on local filesystem will fail. So we need to include directories along the mount path as well.